edgePy/DGEList.py

```diff
@@ -272,37 +306,120 @@ def library_size(self) -> np.array:
         """
         return np.sum(self.counts, 0)
 
-    def cpm(self, log: bool = False, prior_count: float = PRIOR_COUNT) -> None:
-        """Return the DGEList normalized to read counts per million."""
-        self.counts = 1e6 * self.counts / np.sum(self.counts, axis=0)
-        if log:
-            self.counts[self.counts == 0] = prior_count
-            self.counts = np.log(self.counts)
+    def log_transform(self, counts, prior_count):
+        """Compute the log of the counts"""
+        counts[counts == 0] = prior_count
+        return np.log(counts)
+
+    def cpm(self, transform_to_log: bool = False, prior_count: float = PRIOR_COUNT) -> "DGEList":
+        """Normalize the DGEList to read counts per million."""
+        counts = 1e6 * self.counts / np.sum(self.counts, axis=0)
+        current_log = self.log
+        if transform_to_log:
+            counts = self.log_transform(counts, prior_count)
+            current_log = True
+            
+        return self.copy(counts=counts, current_log=current_log)    
 
     def rpkm(
-        self, gene_lengths: Mapping, log: bool = False, prior_count: float = PRIOR_COUNT
-    ) -> None:
+        self, gene_data: ImportCanonicalData, log: bool = False, prior_count: float = PRIOR_COUNT
+    ) -> "DGEList":
         """Return the DGEList normalized to reads per kilobase of gene length
-        per million reads.
+        per million reads. (RPKM =   numReads / ( geneLength/1000 * totalNumReads/1,000,000 )
 
         """
-        raise NotImplementedError
-
-        # TODO: Implement here
-        # self = self.cpm(log=log, prior_count=prior_count)
+        current_log = self.log
+        temp_gene_len = []
+
+        if self.log:
```

Comment:
Using log(exp()) can affect the data quality numerically. Is it really not an option to keep raw data?